### PR TITLE
chore: remove AttnetsService

### PR DESCRIPTION
**Pull request by twoeths**
**Motivation**

Remove `AttnetsService` so that we don't have to maintain it anymore

**Description**

- Since v1.10, we use the new `DLLAttnetsService` by default so no need to maintain the legacy `AttnetsService` anymore

part of #6578

### CODE REVIEW

File: packages/beacon-node/src/network/core/metrics.ts [hunk 1]
```diff
@@ -214,12 +213,12 @@ export function createNetworkCoreMetrics(register: RegistryMetricCreator) {
...
         help: "Count of long lived subscriptions",
       }),
-      subscribeSubnets: register.gauge<{subnet: number; src: SubnetSource | DLLSubnetSource}>({
+      subscribeSubnets: register.gauge<{subnet: number; src: DLLSubnetSource}>({
         name: "lodestar_attnets_service_subscribe_subnets_total",
         help: "Count of subscribe_subnets calls",
         labelNames: ["subnet", "src"],
       }),
-      unsubscribeSubnets: register.gauge<{subnet: number; src: SubnetSource | DLLSubnetSource}>({
+      unsubscribeSubnets: register.gauge<{subnet: number; src: DLLSubnetSource}>({
```
Comment (nflaig) [inline packages/beacon-node/src/network/core/metrics.ts:221]: might wanna rename this now as there is only one type of subnet now

```suggestion
      unsubscribeSubnets: register.gauge<{subnet: number; src: SubnetSource}>({
```

File: packages/beacon-node/src/network/core/networkCore.ts [hunk 2]
```diff
@@ -197,9 +196,7 @@ export class NetworkCore implements INetworkCore {
 
     const enr = opts.discv5?.enr;
     const nodeId = enr ? fromHexString(ENR.decodeTxt(enr).nodeId) : null;
-    const attnetsService = opts.deterministicLongLivedAttnets
-      ? new DLLAttnetsService(config, clock, gossip, metadata, logger, metrics, nodeId, opts)
-      : new AttnetsService(config, clock, gossip, metadata, logger, metrics, opts);
+    const attnetsService = new DLLAttnetsService(config, clock, gossip, metadata, logger, metrics, nodeId, opts);
```
Comment (nflaig) [inline packages/beacon-node/src/network/core/networkCore.ts:199]: same here, we can keep the naming but as DLL subnets are the de factor standard now, could consider removing prefix

**Review by nflaig — COMMENTED** [review]
LGTM, just some thoughts about the naming going forward

**Review by nflaig — COMMENTED** [review]
the diff is pretty interesting now if you wanna compare the difference between prev. attnet service and dll attnet service but I am wondering if we should rather remove the last commit in this PR and create a follow up PR instead

**Review by nflaig — APPROVED** [review]

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.18.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.18.0) :tada:

**Pull request merged**